docs/book/src/tasks/experimental-features/machineset-preflight-checks.md

```diff
@@ -1,12 +1,64 @@
 # Experimental Feature: MachineSetPreflightChecks (alpha)
 
-The `MachineSetPreflightChecks` feature can provide additional safety while creating new Machines for a MachineSet.
+The `MachineSetPreflightChecks` feature can provide additional safety while creating new Machines and remediating existing unhealthy Machines of a MachineSet.
+
+When a MachineSet creates machines under certain circumstances, the operation fails or leads to a new machine that will be deleted and recreated in a short timeframe,
+leading to unwanted Machine churn. Some of these circumstances include, but not limited to, creating a new Machine when Kubernetes version skew could be violated or 
+joining a Machine when the Control Plane is upgrading leading to failure because of unclear kube-apiserver version. 
+
+Enabling `MachineSetPreflightChecks` provides safety in such circumstances by making sure that a Machine is only created when it is safe to do so.
+
 
 **Feature gate name**: `MachineSetPreflightChecks`
 
 **Variable name to enable/disable the feature gate**: `EXP_MACHINE_SET_PREFLIGHT_CHECKS`
 
-The following preflight checks are performed when the feature is enabled:
-* ControlPlaneIsStable
-* KubeadmVersionSkew
-* KubernetesVersionSkew
\ No newline at end of file
+## Supported PreflightChecks
+
+### `ControlPlaneIsStable`
+
+* This preflight check ensures that the ControlPlane is currently stable i.e. the ControlPlane is currently neither provisioning nor upgrading.   
+* This preflight check is only performed if:
+  * The Cluster uses a ControlPlane provider.
+  * ControlPlane version is defined (`controlPlane.spec.version` is set).
```

Comment:
Sorry my previous comment was a bit misleading. What I was trying to say is let's use the path "as the user sees it".
The field paths are fine now but I would continue to write the CR name the way users write it in YAML
`ControlPlane.spec.version`
`MachineSet.spec.template.spec.version`

(ControlPlane obviously doesn't exist but I would still go with the normal naming convention, cf. KubeadmControlPlane)

Feel free to squash after this one